src/main/frontend/components/block.cljs

```diff
@@ -2453,31 +2453,33 @@
 
 (defn- trigger-custom-query!
   [state]
-  (let [[config query] (:rum/args state)
-        repo (state/get-current-repo)
-        result-atom (atom nil)
-        query-atom (if (:dsl-query? config)
-                     (let [q (:query query)
-                           form (safe-read-string q false)]
-                       (cond
-                          ;; Searches like 'foo' or 'foo bar' come back as symbols
-                         ;; and are meant to go directly to full text search
-                         (and (util/electron?) (symbol? form)) ; full-text search
-                         (p/let [blocks (search/block-search repo (string/trim (str form)) {:limit 30})]
-                           (when (seq blocks)
-                             (let [result (db/pull-many (state/get-current-repo) '[*] (map (fn [b] [:block/uuid (uuid (:block/uuid b))]) blocks))]
-                               (reset! result-atom result))))
-
-                         (symbol? form)
-                         (atom nil)
-
-                         :else
-                         (query-dsl/query (state/get-current-repo) q)))
-                     (db/custom-query query))
-        query-atom (if (instance? Atom query-atom)
-                     query-atom
-                     result-atom)]
-    (assoc state :query-atom query-atom)))
+  (let [[config query] (:rum/args state)]
+    (if (or (:custom-query? config) (:dsl-query? config))
```

Comment:
Custom queries can still be nested here before because this function will be called during `will-mount`.